internal/controller/pipelinerollout_controller.go

```diff
@@ -280,14 +282,24 @@ func (r *PipelineRolloutReconciler) reconcile(
 		controllerutil.AddFinalizer(pipelineRollout, finalizerName)
 	}
 
+	var pipelineSpec struct {
+		InterStepBufferServiceName string `json:"interStepBufferServiceName"`
+	}
+	if err := json.Unmarshal(pipelineRollout.Spec.Pipeline.Raw, &pipelineSpec); err != nil {
+		return false, fmt.Errorf("failed to unmarshal pipeline spec: %v", err)
+	}
+	labelKey := "isbsvc-name"
```

Comment:
Could you make a function that given a PipelineRollout returns a map of labels? In this case the map will only have the isbs-name but we could add more logic to the function to add more labels as needed. Then, you can call that func to set the labels of the newPipelineDef object.